notify/notify.go

```diff
@@ -518,16 +518,24 @@ func NewMuteStage(m types.Muter) *MuteStage {
 }
 
 // Exec implements the Stage interface.
-func (n *MuteStage) Exec(ctx context.Context, _ log.Logger, alerts ...*types.Alert) (context.Context, []*types.Alert, error) {
-	var filtered []*types.Alert
+func (n *MuteStage) Exec(ctx context.Context, logger log.Logger, alerts ...*types.Alert) (context.Context, []*types.Alert, error) {
+	var (
+		filtered []*types.Alert
+		muted    []*types.Alert
+	)
 	for _, a := range alerts {
 		// TODO(fabxc): increment total alerts counter.
 		// Do not send the alert if muted.
-		if !n.muter.Mutes(a.Labels) {
+		if n.muter.Mutes(a.Labels) {
+			muted = append(muted, a)
+		} else {
 			filtered = append(filtered, a)
 		}
 		// TODO(fabxc): increment muted alerts counter if muted.
 	}
+	if len(muted) > 0 {
+		level.Debug(logger).Log("msg", "Notifications will not be sent for muted alerts", "alerts", fmt.Sprintf("%v", muted))
```

Comment:
Sorry, I can't dismiss my approval, but this needs a change.

I think we should truncate the alert list to a maximum - I could have swore there are others places where we do this but I couldn't find them.

Reply:
Can you share what your main concerns are?

This would be the first place where we truncate the alert slice when logging in debug mode. The main place this happens untruncated is [here](https://github.com/prometheus/alertmanager/blob/280a742b766d0b4079b02be100cf9541e29f0a20/dispatch/dispatch.go#L515) and [here](https://github.com/prometheus/alertmanager/pull/3438).

If we do truncate it for silences, we should also truncate it elsewhere. However if we do that then debug logs become rather useless for debugging missing notifications as there is no other record of which alerts are in which aggregation groups.

Reply:
You're right - I'm probably misremembering. 

My main concern was debug being too spammy but I this is a very valid use case.